rcl/include/rcl/publisher.h

```diff
@@ -37,6 +37,8 @@ typedef struct rcl_publisher_s
 {
   /// Pointer to the publisher implementation
   rcl_publisher_impl_t * impl;
+  /// Disable flag to loan messages
+  bool disable_loaned_message;
```

Comment:
Or maybe this can be in `rcl_publisher_option`? currently we do not have this option via rclcpp, so integrated in `rcl` layer with environmental variable would be good enough?

Reply:
It would be nice to make this a publisher option, so we *could* control it via rclcpp if we wanted in the future.  I'd vote for moving it there.  Same goes for the subscription.

Reply:
okay, so we will do.

- extend option for publisher and subscription that can be set by user application via rclcpp.
- environmental variable check during initialization for publisher and subscription, and if environmental disable flag is on, that prevails option setting in the argument.

Reply:
@fujitatomoya @clalancette I agree and it was my original proposal to add `disable_loaned_message` to the `rcl_publisher_options_t` and `rcl_subscription_options_t`.
Perhaps we can read out environment variable and init `disable_loaned_message` in `rcl_publisher_get_default_options()`
https://github.com/ros2/rcl/blob/d651a5cc77d3b81b1d779e611cae651238960206/rcl/src/rcl/publisher.c#L197-L207
and in `rcl_subscription_get_default_options()` https://github.com/ros2/rcl/blob/d651a5cc77d3b81b1d779e611cae651238960206/rcl/src/rcl/subscription.c#L200-L210
respectively.

Reply:
addressed by https://github.com/ros2/rcl/pull/1029/commits/3b25f4b8fa1043d8c83c490d2d6d6c41f1780132